taxonomy/models.py

```diff
@@ -357,3 +357,84 @@ def __repr__(self):
             'unique_postings="{4!r} unique_companies={5!r}">'.format(
                 self.id, self.job, self.median_salary, self.median_posting_duration, self.unique_postings,
                 self.unique_companies)
+
+
+class Translations(TimeStampedModel):
+    """
+    Model to save translated descriptions.
+
+    .. no_pii:
+    """
+
+    source_model_name = models.CharField(
+        max_length=255,
+        help_text=_(
+            'The name of the model to which the source belongs e:g Course.'
+        )
+    )
+
+    source_model_field = models.CharField(
+        max_length=255,
+        help_text=_(
+            'The name of the source field to be translated e:g course description.'
+        )
+    )
+
+    source_record_identifier = models.CharField(
+        max_length=255,
+        unique=True,
+        help_text=_('The identifier of the source record e:g course key.')
+    )
+
+    source_text = models.TextField(
+        blank=True,
+        null=True,
+        help_text=_(
+            'The source text to be translated.'
+        )
+    )
+
+    source_language = models.CharField(
```

Comment:
languages should be 2 char long https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes. Can we add choices here?

Reply:
As per this reference https://cloud.google.com/translate/docs/languages Google Client is using multiple formats which could be 2-8 characters long so we can't use "choices" here. I have set up the length equal to 8 for both language fields.